content/200-concepts/100-components/01-prisma-schema/08-views.mdx

```diff
@@ -0,0 +1,180 @@
+hidePage: false
+preview: true
+tocDepth: 2
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+Support for views is currently a very early, experimental preview feature. You can add a view to your Prisma schema with the `view` keyword, but you cannot introspect your schema with `db pull` or apply your schema to your database with Prisma Migrate and `db push`.<br /><br />For updates on progress with this feature, follow [our GitHub issue](https://github.com/prisma/prisma/issues/17335).
+
+</Admonition>
+
+Database views allow you to name and store queries. In relational databases, views are [stored SQL queries](https://www.postgresql.org/docs/current/sql-createview.html) that might include columns in multiple tables, or calculated values such as aggregates. In MongoDB, views are queryable objects where the contents are defined by an [aggregation pipeline](https://www.mongodb.com/docs/manual/core/aggregation-pipeline) on other collections.
+
+This guide explains how to enable the `views` preview feature and add views to your Prisma schema.
+
+</TopBlock>
+
+## How to enable the <inlineCode>views</inlineCode> preview feature
+
+Support for views is currently in an early preview. To enable the `views` preview feature, add the `views` feature flag to the `previewFeatures` field of the `generator` block in your Prisma schema file:
+
+```prisma file=schema.prisma highlight=3;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["views"]
+}
+```
+
+## How to add views to your Prisma schema
+
+To add a view to your Prisma schema, use the `view` keyword. For example, take the following `User` model with a related `Profile` model:
+
+<TabbedContent tabs={[<FileWithIcon text="Relational databases" icon="database"/>, <FileWithIcon text="MongoDB" icon="database"/>]}>
+<tab>
+
+```prisma
+model User {
+  id      Int      @id @default(autoincrement())
+  email   String   @unique
+  name    String?
+  profile Profile?
+}
+
+model Profile {
+  id     Int    @id @default(autoincrement())
+  bio    String
+  user   User   @relation(fields: [userId], references: [id])
+  userId Int    @unique
+}
+```
+
+</tab>
+<tab>
+
+```prisma
+model User {
+  id      String   @id @default(auto()) @map("_id") @db.ObjectId
+  email   String   @unique
+  name    String?
+  profile Profile?
+}
+
+model Profile {
+  id     String @id @default(auto()) @map("_id") @db.ObjectId
+  bio    String
+  User   User   @relation(fields: [userId], references: [id])
+  userId String @unique @db.ObjectId
+}
+```
+
+</tab>
+</TabbedContent>
+
+Next, take a `UserInfo` view in the underlying database that combines the `email` and `name` fields from the `User` model and the `bio` field from the `Profile` model.
+
+For a relational database, the SQL statement to create this view is:
+
+```sql
+CREATE VIEW "UserInfo" AS
+    SELECT u.id, email, name, bio
+    FROM "User" u
+    LEFT JOIN "Profile" p ON u.id = p."userId"
+```
```

Comment:
For MySQL I actually needed to run this:
```sql
CREATE VIEW UserInfo AS
	    SELECT u.id, email, name, p.bio
	    FROM User u
	    LEFT JOIN Profile p ON u.id = p.userId;
```

Reply:
Flavian also did something like this for MySQL: https://github.com/prisma/prisma-engines/pull/3581/files#diff-22acb9f8d5a6ffc5bc3110715c576543886376b5e1b3c929fa2059701a2f327dR141-R154

Reply:
For PostgreSQL it didn't work until I added the quotes

Reply:
So you did have to include `p.bio` and not `bio`?

Reply:
No, this comment is not about `p.bio` - that change makes not difference.
This is about not having double quotes aroung the field names. 
That can not be run on my MySQL I was testing with even.
And if it runs, it might lead to the Client queries not actually being able to find the view.

Reply:
Will investigate under: https://github.com/prisma/docs/issues/4418